util/file.go

```diff
@@ -117,9 +117,15 @@ func GlobCanonicalPath(basePath string, globPaths ...string) ([]string, error) {
 			globPath = filepath.Join(basePath, globPath)
 		}
 
+		const stackDir = "/.terragrunt-stack/"
+
 		matches, err := zglob.Glob(globPath)
 		if err == nil {
 			paths = append(paths, matches...)
+		} else if errors.Is(err, os.ErrNotExist) && strings.Contains(CleanPath(globPath), stackDir) {
```

Comment:
It's odd to me that we're adjusting `GlobCanonicalPath` like this. I feel like there should either be options made available to alter the behavior of this, or that a separate function should be used that signals awareness of special directories like `.terragrunt-stack` directories.